test/integration/test_create_dss_docker_image.py

```diff
@@ -167,3 +168,83 @@ def test_docker_socket_on_host_touched(dss_container_context, fake_docker_socket
         wait_for_socket_access(container)
 
     assert stat_before == socket.stat()
+
+
+class SocketManipulator:
+    def __init__(self, context_provider, socket_on_host: Path):
+        self._context_provider = context_provider
+        self._socket_on_host = socket_on_host
+        self._container = None
+
+    @contextmanager
+    def context(self):
+        with self._context_provider(self._socket_on_host) as container:
+            self._container = container
+            yield self
+            self._container = None
+
+    def run(self, command: str, **kwargs) -> str:
+        """
+        Execute command in container and verify success.
+        """
+        exit_code, output = self._container.exec_run(command, **kwargs)
+        output = output.decode("utf-8").strip()
+        assert exit_code == 0, output
+        return output
+
+    def numeric_gid(self, group_entry: str) -> int:
+        """group_entry is "ubuntu:x:971:", for example"""
+        return int(group_entry.split(':')[2])
+
+    def get_gid(self, group_name: str) -> int:
+        output = self.run(f"getent group {group_name}")
+        return self.numeric_gid(output)
+
+    def get_unassigned_gid(self) -> int:
+        """
+        Return a new gid, that is not used for any other group, yet.
+        """
+        gid = 0
+        for line in self.run("cat /etc/group").splitlines():
+            if not line.startswith(f"nogroup:"):
+                gid = max(gid, self.numeric_gid(line))
+        return gid + 1
+
+    def chgrp_of_docker_socket_on_host(self, gid: str):
+        self.run(f"chgrp {gid} {DOCKER_SOCKET_CONTAINER}", user="root")
```

Comment:
The chgrp should not happen in the container under test, because the entry point might have already ran

Reply:
Yeah, it is probably better to use a different container (Ubuntu image is fine). This way we can setup everything before the entry point is running and we test what we expect and avoid race conditions

Reply:
implemented